Source/WebCore/dom/Observable.cpp

```diff
@@ -169,6 +171,16 @@ void Observable::some(ScriptExecutionContext& context, Ref<PredicateCallback>&&
     return createInternalObserverOperatorSome(context, *this, WTFMove(callback), options, WTFMove(promise));
 }
 
+void Observable::reduce(ScriptExecutionContext& context, Ref<ReducerCallback>&& callback, JSC::JSValue initialValue, const SubscribeOptions& options, Ref<DeferredPromise>&& promise)
+{
+    // IDL for 'optional any' defaults to 'undefined'.
+    // We need to use std::optional to distinguish between 'undefined' and 'not provided'.
+    std::optional<JSC::JSValue> internalInitialValue;
```

Comment:
This is mainly so the reducer does not call for the first value, if the initial value is undefined/not passed. If we did not use std::optional, we'd have more machinery to detect an reducer that returns undefined (this setting the accumulator) and an initial value that is undefined.

So think—and could be wrong—that the semantics of our operator constructor, should have a concept of set and unset initial values; thus using std::optional.

Reply:
To prove this we need to add test cases that stress test these edge cases and fine distinctions. Reasoning about the source code won’t be enough to get this right.